### PR TITLE
Simplify the user interface

### PR DESCRIPTION
**Pull request by frapac**
- add new constructors to SPModel types
- rename modules to be more explicit
- update dams.jl and damsvalley.jl
- improve coverage of unit-testings
- add a test based upon extended formulation

### CODE REVIEW

File: src/extensiveFormulation.jl [hunk 1]
```diff
@@ -7,47 +7,46 @@
...
     #Define the variables for the extensive formulation
     #At each node, we have as many variables as nodes
     @defVar(mod,  u[t=1:T,n=1:DIM_CONTROL*N[t+1]])
     @defVar(mod,  x[t=1:T+1,n=1:DIM_STATE*N[t]])
     @defVar(mod,  c[t=1:T,n=1:N_NOISES*N[t]])
-    
-    
+
+
     #Define the conditional probabilities on each arc of the scenario tree
```
Comment (leclere) [inline src/extensiveFormulation.jl]: No, it computes the total probability of each node from the conditional probabilities.

File: src/noises.jl [hunk 2]
```diff
@@ -83,14 +85,14 @@ Parameters:
     Vector of discrete independent random variables
 
 - n::Int
-    number of simulations computed
+    number of simulations to compute
```
Comment (leclere) [inline src/noises.jl:88]: scenarios

File: src/noises.jl [hunk 3]
```diff
@@ -83,14 +85,14 @@ Parameters:
...
 
 
 Returns :
 - scenarios Array(Float64,n,T)
     an Array of scenario, scenarios[i,:] being the ith noise scenario
 """
-function simulate(law::Vector{NoiseLaw}, n::Int64)
+function generate_scenarios(law::Vector{NoiseLaw}, n::Int64)
     if n <= 0
         error("negative number of simulations")
```
Comment (leclere) [inline src/noises.jl:97]: scenarios

File: test/runtests.jl [hunk 4]
```diff
@@ -80,25 +87,142 @@ facts("SDDP algorithm") do
...
+        # Test upper bounds estimation with Monte-Carlo:
+        n_simulations = 100
+        upb = StochDynamicProgramming.estimate_upper_bound(model, params, V, pbs,
+                                                           n_simulations)[1]
+        @fact typeof(upb) --> Float64
+
+        # Test a simulation upon given scenarios:
+        aleas = simulate_scenarios(model.noises,
+                              (model.stageNumber,
+                               params.forwardPassNumber,
```
Comment (leclere) [inline test/runtests.jl:123]: Use n_simulations instead, else you are checking a monte carlo result with 10 simulation.

File: test/runtests.jl [hunk 5]
```diff
@@ -80,25 +87,142 @@ facts("SDDP algorithm") do
...
+                               model.dimNoises))
+
+        sddp_costs, stocks = forward_simulations(model, params, V, pbs, aleas)
+
+        # Compare sddp cost with those given by extensive formulation:
+        ef_cost = StochDynamicProgramming.extensive_formulation(model,params)
+        @fact typeof(ef_cost) --> Float64
+
+        @fact mean(sddp_costs) --> roughly(ef_cost)
```
Comment (leclere) [inline test/runtests.jl:132]: Checking the lower bound might be more reliable.

**Comment by leclere** [comment]
Good job. A few points to modify and we can merge.

File: src/extensiveFormulation.jl (hunk 1 above)
Comment (frapac) [inline src/extensiveFormulation.jl]: You should see that with Henri.

**Comment by frapac** [comment]
I just rescope this PR thank to your comments. 
If you are ok, we coud merge

**Pull request merged**